File: src/01-easy/the-river-ii-/solution.cs
using System;
class Solution {
    static int Ds(long n){int s=0;while(n>0){s+=(int)(n%10);n/=10;}return s;}
    static void Main(){
        long r=long.Parse(Console.ReadLine());
        long lo=Math.Max(1,r-45);
        for(long x=lo;x<r;x++)if(x+Ds(x)==r){Console.WriteLine("YES");return;}
        Console.WriteLine("NO");
    }
}

File: src/01-easy/the-river-ii-/solution.vb
Imports System
Module Solution
    Function Ds(ByVal n As Long) As Long
        Dim s As Long = 0
        While n > 0
            s += n Mod 10
            n = n \ 10
        End While
        Return s
    End Function
    Sub Main()
        Dim r As Long = Long.Parse(Console.ReadLine())
        Dim lo As Long = Math.Max(1, r - 45)
        For x As Long = lo To r - 1
            If x + Ds(x) = r Then
                Console.WriteLine("YES")
                Return
            End If
        Next
        Console.WriteLine("NO")
    End Sub
End Module
